### PR TITLE
feat(core): Identifier in NanoTDF ResourceLocator borrowed from Protocol

### PR DESCRIPTION
**Pull request by pflynn-virtru**
Add support for identifiers in ResourceLocator

Introduced an IdentifierType enum to categorize different identifier lengths. Updated ResourceLocator constructors to handle identifiers, including parsing and setting appropriate identifier types. Modified ByteBuffer constructor to extract identifier information from the protocol byte.

Issue: https://github.com/opentdf/platform/issues/1203
Specification: https://github.com/opentdf/spec/pull/40
ADR: https://github.com/opentdf/platform/issues/900

### CODE REVIEW

File: sdk/src/main/java/io/opentdf/platform/sdk/nanotdf/ResourceLocator.java [hunk 1]
```diff
@@ -73,17 +147,32 @@ public int writeIntoBuffer(ByteBuffer buffer) {
...
         totalBytesWritten += 1;
 
-        // Write the url body;
+        // Write the url body
         buffer.put(body);
         totalBytesWritten += body.length;
 
+        // Write the identifier
+        if (identifierType != NanoTDFType.IdentifierType.NONE) {
+            buffer.put(identifier);
```
Comment (dmihalcik-virtru) [inline sdk/src/main/java/io/opentdf/platform/sdk/nanotdf/ResourceLocator.java:168]: You may need to pad the buffer here or assert (throw IllegalArg) if identifier is. the wrong length

Comment (pflynn-virtru) [reply sdk/src/main/java/io/opentdf/platform/sdk/nanotdf/ResourceLocator.java:169]: Note we have this setter (without getter)
```
public void setProtocol()
```

Comment (pflynn-virtru) [reply sdk/src/main/java/io/opentdf/platform/sdk/nanotdf/ResourceLocator.java:204]: done added padded and tests

File: sdk/src/main/java/io/opentdf/platform/sdk/nanotdf/ResourceLocator.java [hunk 2]
```diff
@@ -1,37 +1,140 @@
...
+                throw new IllegalArgumentException("Unexpected identifier type: " + identifierType);
+        }
+    }
+
+    public void setIdentifier(String identifier) {
+        if (identifier == null) {
+            this.identifierType = NanoTDFType.IdentifierType.NONE;
+            this.identifier = new byte[0];
+        } else {
+            byte[] identifierBytes = identifier.getBytes();
```
Comment (mkleene) [inline sdk/src/main/java/io/opentdf/platform/sdk/nanotdf/ResourceLocator.java:116]: I think we should set the charset here. If we [don't it uses the platform default](https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#getBytes--)

**Review by mkleene — COMMENTED** [review]
Looks good except for the one comment!

**Comment by dmihalcik-virtru** [comment]
merged as https://github.com/opentdf/java-sdk/pull/112

**Pull request closed without merging**